docx/oxml/numbering.py

```diff
@@ -139,11 +139,21 @@ def get_abstractNum(self, numId):
             if el.abstractNumId == abstractNum_id:
                 return el
 
-    def get_lvl_for_p(self, p, styles_cache):
+    def get_lvl_from_props(self, p, styles_cache=None):
         """
-        Gets the formatting based on current paragraph indentation level.
+        Gets the formatting based on current paragraph indentation level defined in paragraph styles.
+        If ``styles_cache`` is not None then level from style formating is fetched otherwise
+        level from direct paragraph formating is used.
         """
-        numPr = p.pPr.get_numPr(styles_cache)
+        try:
+            if styles_cache:
+                numPr = p.pPr.get_style_numPr(styles_cache)
+            else:
+                numPr = p.pPr.numPr
```

Comment:
this seems like a good candidate for https://realpython.com/python-conditional-statements/#one-line-if-statements